### PR TITLE
Add repo rule for fetching a Python SDK

### PR DESCRIPTION
**Pull request by JIoJIaJIu**
Backbone for https://github.com/bazelbuild/bazel/issues/7467
* Doesn't support Windows
* I haven't covered it with tests due to [the issue](https://stackoverflow.com/questions/55900936/long-runtime-for-build-run-test-py-test-executed-with-testbase-runbazel), shall I raise a ticket? What is the best approach to apply the tests for the implementation?
* Periodically I see `Restarting` after `Install` phase, what can be the cause?

### CODE REVIEW

**Comment by Globegitter** [comment]
As much as I love seeing this, would it maybe make more sense to have this in an external repository? https://github.com/bazelbuild/rules_python/ has not received much love recently but imo it would probably be the best place for this to have fixes/updates to this repo rule independent of bazel releases.

**Comment by JIoJIaJIu** [comment]
Hi @Globegitter 
Yes, it sounds reasonable, I've created the PR here cause `py toolchain` is a part of `bazel_tools`and the issue #7467 was created here. But looks like there is no anything that doesn't allow to move it out to `python_rules`

What do you think, @brandjon 

Could you please also clarify my doubts above related testing
FYI as a part of testing approach I decided to create mocked python tar based on `autoconf` which simulates configuration/building and installing python and would like to upload it into the source code as done with [six package](https://github.com/bazelbuild/bazel/tree/52af3ccaaa75d0ee95c83206abeb5fa925903bf0/src/test/py/bazel/testdata/bazel_external_repository_test). am I looking in a right direction or not?

**Comment by Globegitter** [comment]
Also quite interesting project that has a standalone python for windows: https://github.com/indygreg/python-build-standalone - though I have not tested it yet.

**Comment by JIoJIaJIu** [comment]
@Globegitter thank you for your sharing!
@brandjon, do you have any updates regarding it? Fill free to close the PR, cause the work should be done in https://github.com/bazelbuild/rules_python/
I would like to proceed work on the issue but needs your feedback

File: tools/python/sdk/sdk.bzl [hunk 1]
```diff
@@ -0,0 +1,112 @@
+def get_binary_url(version):
+    arch = "Python-{}.tar.xz".format(version)
+    return "https://www.python.org/ftp/python/{}/{}".format(version, arch)
+
+def configure(ctx):
+    ctx.report_progress("Configure")
+    res = ctx.execute([
+        "./configure",
+        "--prefix=" + str(ctx.path("./"))
```
Comment (keith) [inline tools/python/sdk/sdk.bzl]: We might want something like `--enable-optimizations` here too?

**Comment by jin** [comment]
Perhaps this should be reopened in rules_python?

**Comment by brandjon** [comment]
Right, this should definitely be reopened in rules_python. I'll transfer the underlying issue.

**Comment by jin** [comment]
Closing this for now - please reopen a new PR in the rules_python repository.

**Comment by ddlees** [comment]
FWIW I ran into this problem and saw it wasn't solved so I just recently published some rules that leverage [pyenv](https://github.com/pyenv/pyenv) to download, compile and register the python toolchain with whatever `py_runtime_pair` the project calls for.

It's probably a little rough but feedback and PRs are welcome; I just hope what's currently there can help some folks. 😉 🍻 

https://github.com/digital-plumbers-union/rules_pyenv

**Pull request closed without merging**